packages/contracts-bedrock/src/L2/L2CrossDomainMessenger.sol

```diff
@@ -15,26 +15,27 @@ import { Constants } from "src/libraries/Constants.sol";
 ///         L2 on the L2 side. Users are generally encouraged to use this contract instead of lower
 ///         level message passing contracts.
 contract L2CrossDomainMessenger is CrossDomainMessenger, ISemver {
-    /// @custom:semver 1.8.0
-    string public constant version = "1.8.0";
+    /// @custom:semver 1.9.0
+    string public constant version = "1.9.0";
 
     /// @notice Constructs the L2CrossDomainMessenger contract.
-    /// @param _l1CrossDomainMessenger Address of the L1CrossDomainMessenger contract.
-    constructor(address _l1CrossDomainMessenger) CrossDomainMessenger(_l1CrossDomainMessenger) {
-        initialize();
+    /// @param _l1CrossDomainMessenger Address of the L1CrossDomainMessenger contract on the other network.
+    constructor(address _l1CrossDomainMessenger) CrossDomainMessenger() {
+        initialize({ _l1CrossDomainMessenger: _l1CrossDomainMessenger });
     }
 
     /// @notice Initializer.
-    function initialize() public initializer {
-        __CrossDomainMessenger_init();
+    /// @param _l1CrossDomainMessenger Address of the L1CrossDomainMessenger contract on the other network.
+    function initialize(address _l1CrossDomainMessenger) public initializer {
```

Comment:
nit: can you update this to be a `CrossDomainMessenger` type instead of `address`?

Reply:
double checking here: this change would clash with the underlying type of `universal/CrossDomainMessenger.sol`. Updating that would have deeper effects. On the other hand, it seems like manually wrapping the address in `CrossDomainMessenger` would require later unwrapping it to use in relation to  `universal/CrossDomainMessenger.sol`, which feels unnecessary. maybe i'm missing something?